lib/shanty/config.rb

```diff
@@ -0,0 +1,51 @@
+require 'shenanigans/hash/to_ostruct'
+require 'deep_merge'
+
+module Shanty
+  # Public: Configuration class for shanty
+  class Config
+    CONFIG_FILE = '.shanty.yml'
+
+    def initialize(root, environment)
+      @root = root
+      @environment = environment
+    end
+
+    def merge!(new_config)
+      @config = config_hash.clone.deep_merge(new_config)
+    end
+
+    def [](key)
+      to_ostruct[key]
+    end
+
+    def method_missing(m)
+      os_resp = to_ostruct.send(m)
+      return os_resp unless os_resp.nil?
+      OpenStruct.new
+    end
+
+    def respond_to?(method_sym, include_private = false)
+      to_ostruct.send.respond_to?(method_sym, include_private)
```

Comment:
You need to also check whether this instance responds to a given method. For example, this will respond `false` to `respond_to?(:merge!)`.

```ruby
super || to_ostruct.send.respond_to?(method_sym, include_private)
```